File: receipt-analyzer-service/src/main/java/pl/madzierski/daniel/app/receipt/revision/model/ReceiptRevisionResolveData.java
package pl.madzierski.daniel.app.receipt.revision.model;

import lombok.Builder;

import java.util.List;

@Builder
public record ReceiptRevisionResolveData(
        String revisionVersion,
        String brand,
        List<ReceiptRevisionResolveDataItem> items,
        List<ReceiptRevisionResolveDataFile> files
) {
    public record ReceiptRevisionResolveDataItem(
            String name,
            Double amount,
            Double unitPrice,
            Double discount,
            Double totalPrice,
            Integer position
    ) {
    }

    public record ReceiptRevisionResolveDataFile(
            String name,
            int page,
            String rawData
    ) {
    }
}

File: receipt-analyzer-service/src/main/kotlin/pl/madzierski/daniel/app/receipt/revision/model/ReceiptRevisionResolveData.kt
package pl.madzierski.daniel.app.receipt.revision.model

data class ReceiptRevisionResolveData(
    var revisionVersion: String?,
    var brand: String?,
    var items: List<ReceiptRevisionResolveDataItem>?,
    var files: List<ReceiptRevisionResolveDataFile>?
) {
    data class ReceiptRevisionResolveDataItem(
        var name: String?,
        var amount: Double?,
        var unitPrice: Double?,
        var discount: Double?,
        var totalPrice: Double?,
        var position: Int?,
    )

    data class ReceiptRevisionResolveDataFile(
        var name: String,
        var page: Int,
        var rawData: String,
    )
}
